nsfc-plugin/src/main/java/org/osc/controller/nsfc/api/NeutronSfcSdnRedirectionApi.java

```diff
@@ -69,15 +77,73 @@ public InspectionPortElement getInspectionPort(InspectionPortElement inspectionP
             return null;
         }
 
-        return null; // TODO (Dmitry) Implement
+        String portPairId = inspectionPort.getElementId();
+        PortPair portPair = null;
+
+        if (portPairId != null) {
+            portPair = this.osCalls.getPortPair(portPairId);
+        }
+
+        if (portPair == null) {
+            LOG.warn("Failed to retrieve InspectionPort by id! Trying by ingress and egress " + inspectionPort);
+
+            NetworkElement ingress = inspectionPort.getIngressPort();
+            NetworkElement egress = inspectionPort.getEgressPort();
+
+            portPair = this.utils.fetchInspectionPortByNetworkElements(ingress, egress);
+        }
+
+        if (portPair != null) {
+            // only id is ever used
+            return new InspectionPortEntity(portPairId, null, null, null);
```

Comment:
the ingress and egress are provided by port pair, so i would pass them on. 

Ideally we get PPG as well, but  since SFC does not make determining this easy lets implement as needed.